lib/takeaway.rb

```diff
@@ -0,0 +1,43 @@
+require './lib/menu.rb'
+require './lib/sms_sender.rb'
+class Takeaway
+
+attr_reader :customer_order
+
+ def view_menu
+  Menu.new.menu
+ end
+
+ def place_order(order_hash, exact_total)
+  @customer_order = order_hash
+  order_checker
+  message = 'sorry your calculated sum does not match the order total'
+  raise(message) if order_sum != exact_total
+  sms_send
+ end
+
+ def order_sum
+  cost = view_menu.map {|k, v| v * @customer_order[k] if @customer_order.key? k}.compact
```

Comment:
Metrics/LineLength: Line is too long. [88/80] (https://github.com/bbatsov/ruby-style-guide#80-character-limits)